### PR TITLE
Use phase correctly

### PR DESCRIPTION
**Pull request by tommyp**
With the changes in https://github.com/alphagov/finder-frontend/pull/230
we now use the `phase` attribute from the top level of the Content Item
hash. This commit modifies the check to merge this if it is present in
the metadata along with adding a test to test this behaviour.

### CODE REVIEW

File: spec/lib/publishing_api_finder_publisher_spec.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@
       }
     end
 
-    def make_metadata base_path, overrides = {}
+    def make_metadata(base_path, overrides = {})
```
Comment (mikejustdoit) [inline spec/lib/publishing_api_finder_publisher_spec.rb]: The change on this line is excellent, but unrelated to the rest of the commit... can we make it a separate one?

Comment (tommyp) [inline spec/lib/publishing_api_finder_publisher_spec.rb]: Ah yeah. Meant to do that in a separate commit but forgot I'd changed it. Added it to an extra commit and pushed it.

**Comment by mikejustdoit** [comment]
Other than the minor comment, looks good!

Nice to see some context in the commit message :+1:
